notebooks/deepLearning/DeepLearning - Deep Vision Classifier.ipynb

```diff
@@ -0,0 +1,350 @@
+{
```

Comment:
Some comments on this file (hard to leave at the right place so ill stick em in here)

1. You have some complex path and mounting stuff, consider using spark.readBinaryFiles, filtering out the content to just get the paths, and then modifying the name to point to mounted storage instead.

2. Some functions you define are camelcase, consider making these snake case because of python land

3. Is the _transform row function used?
4. in the definition of the udf you dont need to use a lambda, can justt pass func in directly
5. The comment "This is an imoportant parameter doesent tell enough to users, consider a brief explanation"
6. Is the dummy callback required? 
7. Might want to make the deep vision classifier have a few more defaults so that we can keep init simple and clean for this demo
8. you do this thing where you have to set the estimator and the model to have different names for "features" any way tto make it so that the code here is re-used and works without setting on the model?
9. It looks like we need a argmax udf before we send it into the evaluator. Would other spark multiclass stuff need this too? If not we should move our API to align with other spark multiclass stuff if possible
10. your top of the cell looks alot like horovod install script, any way to just use this script directly?

Reply:
1.2.4.6.7.10. Done.
3. Yes, it's used in DeepVisionClassifier's transformation_fn parameter, so that it transforms the image internally.
5. I'll remove this parameter and I haven't figured out how this param should be set tbh.. If I left it as default value in databricks it triggers errors.
8. That's because only estimator accepts transformation_fn so we can use path as input, and model reads the features directly so I use another udf to transform the path into features before inference.
9. I think it's needed as our model output contains an array of label probability instead of the real label prediction.